packages/react-aria-components/src/TableLayout.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {TableLayout as BaseTableLayout, TableLayoutProps} from '@react-stately/layout';
+import {LayoutOptionsDelegate} from './Virtualizer';
+import {TableColumnResizeStateContext} from './Table';
+import {useContext, useMemo} from 'react';
+
+export class TableLayout<T> extends BaseTableLayout<T> implements LayoutOptionsDelegate<TableLayoutProps> {
```

Comment:
Since this is using the BaseTableLayout, I found that it doesn't handle the "loader" type node that the load more spinner collection element due to https://github.com/adobe/react-spectrum/blob/535e1596d2f00c3e30cab2a64106435fdb696f01/packages/%40react-stately/layout/src/TableLayout.ts#L290-L302 not understanding what that node type is. Bit torn on whether or not the TableLayout should be updated to handle "loader" type nodes or if the leaf component for the loader should be changed to "item" (this would pose some difficulties in differentiating the loaders from normal rows but can probably be handled collection side some how)

Or would the expectation be that the TableLayout be replaced/extended to handle new node types? Thinking from an angle where perhaps a section header/loader would need its calculated row height differentiated from a typical row

Reply:
If they were all items, how would we differentiate them elsewhere? Do we need to?

Otherwise I think it's fine to add more item types to the layout, would just work the same as item. I don't expect we'd be adding them that often.

Reply:
For now I've made the loader node trigger `this.buildRow` as well. I do still wonder if there should be additional customization allowed for setting the height of the loader row independent of the row height but that can be handled in the future if requested.